### ISSUE TITLE
Ability to pass parameters and callbacks in sorting

### ISSUE THREAD

**Issue by brajmohans** [yajra/laravel-datatables#340]
First of all thanks for this awesome library. This works great overall but I just came across a limitation.

Currently there is no way to custom sort or pass extra parameters in Laravel collection sort.
As we can pass a callback to `$collection->sortBy()`

But in cases where we want to sort in natural order or case insensitive sort we use `$collection->sortBy` like

```php
$collection->sortBy(function ($field_name, SORT_NATURAL|SORT_FLAG_CASE))
```
or we can pass our callback also.

I can see that there is a filter function which just disables default filtering and override with our callback.
I wonder if something similar can be done with this also by using same method.

This is ordering function from CollectionEngine
```php
public function ordering()
    {
        foreach ($this->request->orderableColumns() as $orderable) {
            $column           = $this->getColumnName($orderable['column']);
            $this->collection = $this->collection->sortBy(
                function ($row) use ($column) {
                    $data = $this->serialize($row);

                    return Arr::get($data, $column);
                }
            );

            if ($orderable['direction'] == 'desc') {
                $this->collection = $this->collection->reverse();
            }
        }
    }
```
As in above snippet there isn't any way to pass callback or extra parameters and there is no order method (just like filter) either.

**Comment by yajra** [comment]
@brajmohans, I do agree that collection engine has very limited functionality and I wasn't able to update the engine because I rarely use this approach. Anyways, your proposal to have something like `filter` function to override ordering is a good idea and I think it's doable. If you can submit a PR, I would gladly accept it. Thanks!

**Comment by yajra** [comment]
@brajmohans, I just added a possible fix. Can you try using `dev-develop` branch and confirm if limitation was addressed? You can override now using `order` method. Thanks! 

```php
->order(function(CollectionEngine $instance) {
// dd($instance->collection);
})
```

**Comment by brajmohans** [comment]
@yajra Thanks for your quick response. I just updated to dev-develop and tried but I had to change ordering function a little bit 
```php
if($this->orderCallback) {
    // return call_user_func($this->orderCallback, $this);
    $this->collection = call_user_func($this->orderCallback,$this); // here updating actual collection
    return;
}
// rest is same
```

**Comment by yajra** [comment]
I just updated dev-develop and set return to void. Can you please check again and will released this on next version when confirmed. Thanks!

**Comment by brajmohans** [comment]
Thing is you are not setting the collection with the sorted value in the user called function like you are doin in the default ordering

```php
    foreach ($this->request->orderableColumns() as $orderable) {
        $column           = $this->getColumnName($orderable['column']);
        $this->collection = $this->collection->sortBy( //note here you are setting the sorted value back to the collection
            function ($row) use ($column) {
                $data = $this->serialize($row);

                return Arr::get($data, $column);
            }
        );

        if ($orderable['direction'] == 'desc') {
            $this->collection = $this->collection->reverse(); //same here
        }
    }
```

But in the update you are just returning after sorting .But not updating the collection instance 

Solution

```php
    if($this->orderCallback) {
        // return call_user_func($this->orderCallback, $this); //setting the back the collection with the sorted value
        $this->collection = call_user_func($this->orderCallback,$this); // here updating actual collection
        return;
    }

    foreach ($this->request->orderableColumns() as $orderable) {
        $column           = $this->getColumnName($orderable['column']);
        $this->collection = $this->collection->sortBy( //note here you are setting the sorted value back to the collection
            function ($row) use ($column) {
                $data = $this->serialize($row);

                return Arr::get($data, $column);
            }
        );

        if ($orderable['direction'] == 'desc') {
            $this->collection = $this->collection->reverse(); //same here
        }
    }
```

**Comment by yajra** [comment]
Oh yeah, overlooked that! But does this kind of snippet does not work? Is it really necessary to return it to collection?

```php
->order(function(CollectionEngine $instance) {
	$instance->collection = $instance->collection->sortBy(
	    function ($row) use ($column) {
	        $data = $instance->serialize($row);

	        return Arr::get($data, $column);
	    }
	);
})

**Comment by brajmohans** [comment]
Yeah its working just fine. Sorry for late response.

**Comment by yajra** [comment]
New order feature released on v6.3.0. Kindly check and confirm. Thanks!
